src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ViewChild,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import { T } from '../../t.const';
+import { FormControl } from '@angular/forms';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { tap, map, take, debounceTime, startWith } from 'rxjs/operators';
+import { TaskService } from '../tasks/task.service';
+import { Router } from '@angular/router';
+import { TODAY_TAG } from '../tag/tag.const';
+import { Project } from '../project/project.model';
+import { Tag } from '../tag/tag.model';
+import { ProjectService } from '../project/project.service';
+import { TagService } from '../tag/tag.service';
+import { Task } from '../tasks/task.model';
+import { blendInOutAnimation } from 'src/app/ui/animations/blend-in-out.ani';
+import { AnimationEvent } from '@angular/animations';
+import { SearchItem, SearchQueryParams } from './search-bar.model';
+import { getWorklogStr } from '../../util/get-work-log-str';
+
+@Component({
+  selector: 'search-bar',
+  templateUrl: './search-bar.component.html',
+  styleUrls: ['./search-bar.component.scss'],
+  animations: [blendInOutAnimation],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchBarComponent implements AfterViewInit, OnDestroy {
+  @Input() isElevated: boolean = false;
+  @Output() blurred: EventEmitter<any> = new EventEmitter();
+
+  @ViewChild('inputEl') inputEl!: ElementRef;
+  @ViewChild('searchForm') searchForm!: ElementRef;
+
+  T: typeof T = T;
+  isLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  taskSuggestionsCtrl: FormControl = new FormControl();
+  filteredIssueSuggestions$: Observable<SearchItem[]> = new Observable();
+  isArchivedTasks: boolean = false;
+  tooManyResults: boolean = false;
+
+  private _attachKeyDownHandlerTimeout?: number;
+  private _tasks: Task[] = [];
+  private _projects: Project[] = [];
+  private _tags: Tag[] = [];
+  private _searchableItems: SearchItem[] = [];
+
+  constructor(
+    private _taskService: TaskService,
+    private _projectService: ProjectService,
+    private _tagService: TagService,
+    private _router: Router,
+  ) {
+    this._taskService.allTasks$.pipe(take(1)).subscribe((t) => (this._tasks = t));
```

Comment:
All these need unsubscribes.

Reply:
I thought take(1) removed the need for unsubscription.  
I could also use toPromise if you prefer, or just unsubscribe explicitly. :)
Note aside, I noticed you didn't unsubscribe from _this.\_subs_ in add-task-bar either. Or does that work differently?
As I said. I'm rather new to all this...

Btw: firstValueFrom() is going to make this look much better :)

Reply:
> I thought take(1) removed the need for unsubscription.

No, because allTasks$ is async. You might run into the (unlikely, but possible) case where the component is unloaded, before a value is emitted. 

Good catch with the add-task-bar. This a bug and potential memory leak. It's not as bad as the component should be loaded only once most of the time, but still, it's generally a good practice to take care of your subscriptions. I'll provide a fix.
 
> I could also use toPromise if you prefer, or just unsubscribe explicitly. :)

I only use toPromise as a convience thing for one time values inside a promise like flow. Here we're dealing with multiple subscriptions. It's better to do this via subscribe/unsubscribe I think.

Reply:
Ok, I think it's best to just always unsubscribe.